prerender/cache/s3.py

```diff
@@ -60,8 +60,6 @@ def set(self, key: str, payload: bytes, ttl: int = None, format: str = 'html') -
         )
 
     def _filename(self, url, format):
-        hex_name = codecs.encode(url.encode('utf-8'), 'hex').decode('utf-8')
-        sub_dir = os.path.join(hex_name[:2], hex_name[2:4])
-        name = hex_name[4:] + '.{}'.format(format)
         parsed_url = urlparse(url)
-        return os.path.join(parsed_url.hostname, sub_dir, name)
+        encoded_name = urlencode(parsed_url.path)
```

Comment:
If `url` has query string and query string may change behaviors of the page, this won't work well.